spring-integration-jms/src/main/java/org/springframework/integration/jms/JmsOutboundGateway.java

```diff
@@ -1109,7 +1113,7 @@ private Object doSendAndReceiveAsync(Destination reqDestination, jakarta.jms.Mes
 				return future;
 			}
 			else {
-				return obtainReplyFromContainer(correlation, replyQueue);
+				return obtainReplyFromContainer(correlation, Objects.requireNonNull(replyQueue));
```

Comment:
Same here: we don't need this assert since when we reach this code block, the `replyQueue` is not null.
The mentioned `@SuppressWarnings` on this method will cover us.